rpcclient/infrastructure.go

```diff
@@ -613,7 +619,7 @@ func (c *Client) resendRequests() {
 	for _, jReq := range resendReqs {
 		// Stop resending commands if the client disconnected again
 		// since the next reconnect will handle them.
-		if c.Disconnected() {
+		if c.balancer.IsAllDisconnected() {
```

Comment:
This is only defined on `RoundRobinBalancer`, so as mentioned elsewhere, this breaks the notion of the code being agnostic to the type of balancer being used.